x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table_toolbar.tsx

```diff
@@ -105,6 +105,7 @@ export const RulesTableToolbar = React.memo(() => {
             getPromptContext={getPromptContext}
             suggestedUserPrompt={i18nAssistant.EXPLAIN_THEN_SUMMARIZE_RULE_DETAILS}
             tooltip={i18nAssistant.RULE_MANAGEMENT_CONTEXT_TOOLTIP}
+            isAssistantEnabled={isAssistantEnabled}
```

Comment:
Can we call `const { hasAssistantPrivilege, isAssistantEnabled } = useAssistantAvailability();` inside the `NewChat` component? Why our code needs to be changed every time some internal logic of the Assistant changes?